### PR TITLE
Darkmode/slot color

### PR DESCRIPTION
**Pull request by JiaqiWang18**
## Description
Style slot colors
image

### CODE REVIEW

File: static/css/timetable/modules/slot.scss [hunk 1]
```diff
@@ -4,19 +4,25 @@
...
-  transition: color .3s, background-color .3s, width .3s, left .3s;
+  transition: color 0.3s, background-color 0.3s, width 0.3s, left 0.3s;
 
   &:hover {
     color: $g222;
   }
 
+  div {
+    span {
+      color: black;
```
Comment (JJamesWWang) [inline static/css/timetable/modules/slot.scss:15]: What does this do?

File: static/js/redux/constants/colors.ts [hunk 2]
```diff
@@ -29,8 +31,7 @@ const COLOUR_DATA = [
   { background: "#E7F76D", highlight: "#C9E20A", border: "#AFC11F", font: "#222" },
   { background: "#A3F5F2", highlight: "#7CD2CF", border: "#53ABA8", font: "#222" },
   { background: "#7499A2", highlight: "#668B94", border: "#6C7A89", font: "#222" },
-  { background: "#E7F76D", highlight: "#C4D44D", border: "#6C7A89", font: "#222" },
```
Comment (JJamesWWang) [inline static/js/redux/constants/colors.ts:32]: How come this got removed?

**Review by JJamesWWang — COMMENTED** [review]
Just some questions before we merge

File: static/js/redux/constants/colors.ts (hunk 2 above)
Comment (JiaqiWang18) [reply static/js/redux/constants/colors.ts:32]: the background color repeats line 31, so it is very hard to distinguish the slots.

**Review by JJamesWWang — CHANGES_REQUESTED** [review]
Not sure if this is possible, but could we prioritize the colors that Lauren chose over the ones they have?

image
I think these colors (6th and after) are uglier than what Lauren has in the style guide

File: static/js/redux/constants/colors.ts (hunk 2 above)
Comment (JJamesWWang) [reply static/js/redux/constants/colors.ts:32]: oh true, good catch

File: static/css/timetable/modules/slot.scss (hunk 1 above)
Comment (JJamesWWang) [reply static/css/timetable/modules/slot.scss:15]: I have figured out what this does lmao

File: static/js/redux/state/slices/themeSlice.ts [hunk 3]
```diff
@@ -1,7 +1,11 @@
+import { RootState } from "./../index";
+import { lightSlotColor, darkSlotColor } from "./../../constants/colors";
 import { createSlice, PayloadAction } from "@reduxjs/toolkit";
+import { SlotColorData } from "../../constants/commonTypes";
 
 interface ThemeSliceState {
   theme: Theme;
+  slotColors: { [k in Theme]?: SlotColorData[] };
```
Comment (JJamesWWang) [inline static/js/redux/state/slices/themeSlice.ts:8]: What is this code lol

**Review by JJamesWWang — APPROVED** [review]
Immaculate

File: static/js/redux/state/slices/themeSlice.ts (hunk 3 above)
Comment (JJamesWWang) [reply static/js/redux/state/slices/themeSlice.ts:8]: Wack

**Pull request merged**